packages/vite/src/generators/configuration/configuration.ts

```diff
@@ -28,27 +30,62 @@ export async function viteConfigurationGenerator(tree: Tree, schema: Schema) {
   );
   let buildTargetName = 'build';
   let serveTargetName = 'serve';
+  let testTargetName = 'test';
 
   schema.includeLib ??= projectType === 'library';
 
+  // This is for when we are convering an existing project
+  // to use the vite executors.
   if (!schema.newProject) {
-    const { buildTarget, serveTarget, unsuppored } =
-      findExistingTargets(targets);
+    const {
+      buildTarget,
+      serveTarget,
+      testTarget,
+      unsupported,
+      unsupportedUserProvidedTarget,
+    } = findExistingTargetsInProject(targets, {
+      build: schema.buildTarget,
+      serve: schema.serveTarget,
+      test: schema.testTarget,
+    });
 
     /**
-     * Here, we make sure that the project has a build target
-     * and it is unsupported, before throwing the error.
-     * The reason is that the project could not have a build
-     * target at all, in which case we don't want to throw.
-     * Or the project can have multiple build targets, one of which unsupported.
-     * In that case, we convert the supported one.
-     */
-    if (!buildTarget && unsuppored) {
+     * This means that we only found unsupported build targets in that project.
+     *
+     * If the `unsupported` flag was false, it would mean that we did not find
+     * a build target at all, so we can create a new one.
+     *
+     * So we only throw if we found a target, but it is unsupported.
+     *
+     * */
+    if (!buildTarget && unsupported) {
       throw new Error(
         `The project ${schema.project} cannot be converted to use the @nrwl/vite executors.`
       );
     }
 
+    if (!unsupported && !buildTarget && !serveTarget && !testTarget) {
+      await handleUnknownExecutors();
+    }
+
+    await handleUnsupportedUserProvidedTargets(
```

Comment:
Nitpick: It's hard to know what the three args here are for. They also don't seem optional, but are declared as such?

Maybe an options interface would be more clear on the intent? Something like this based off of what I understand from the code.

```ts
handleUnsupportedUserProvidedTargets({
  unsupportedUserProvidedTarget,
  userProvidedTargets: {
    build: schema.buildTarget,
    serve: schema.serveTarget,
    test: schema.testTarget,
  },
  defaultTargets: {
    build: buildTargetName,
    serve: serveTargetName,
    test: testTarget,
  }
});
```

Reply:
You are correct. I refactored the names and the interfaces and added some comments. But hopefully the naming of the variables and interfaces should be most descriptive now of what I'm trying to achieve.